.github/workflows/pull-request.yml

```diff
@@ -58,6 +58,37 @@ jobs:
       - name: Run Typescript Type-Checker
         run: npm run typecheck
 
+  File-count-check:
+    name: Counting files changed
+    runs-on: ubuntu-latest
+    env:
+      BASE_BRANCH: ${{ github.event.pull_request.base.ref }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        
+      - name: Checkout code
+        run: git fetch origin pull/${{ github.event.pull_request.number }}/head:pr-branch
+        working-directory: ${{ github.workspace }}
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11.6'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black pydocstyle
+        
+      - name: Get changed files in base branch, compare difference, and execute python script
+        id: changed-files-base
+        run: |
+          git fetch origin $BASE_BRANCH
```

Comment:
Include this logic within the script. No exporting of output to a file.

Reply:
Use subprocess.Popen.communicate it gives you a lot of options for getting the output of OS commands and specific error handling
- https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate